tests/InternalClient.php

```diff
@@ -35,6 +46,21 @@ public function __construct(Container $container, $baseUrl = '/api/v2') {
     }
 
     /**
+     * Add the transient key cookie header to an array of headers.
+     *
+     * @param array $headers
+     */
+    public function addTransientKeyHeader(array &$headers) {
```

Comment:
I'm really not a big fan of all these functions modifying parameters by references instead of returning the modified object.
We are gonna end up with 2 different ways to do things and I that's not great.
Not having to think twice and always assign back the variable to itself when calling the functions is simpler.

Also, passing by reference does is not an optimization at all.